core/src/main/kotlin/fr/sncf/osrd/utils/JavaInteroperabilityTools.kt

```diff
@@ -0,0 +1,20 @@
+package fr.sncf.osrd.utils
+
+import fr.sncf.osrd.sim_infra.api.Route
+import fr.sncf.osrd.sim_infra.api.RouteId
+import fr.sncf.osrd.utils.indexing.StaticIdxList
+import fr.sncf.osrd.utils.indexing.mutableStaticIdxArrayListOf
+import fr.sncf.osrd.utils.units.Speed
+
+/** Java can't handle generics with value classes, it's impossible to extract the underlying speed from a
+ * `RangeMapEntry` in pure java. This method handles this specific case.  */
+fun rangeMapEntryToSpeed(entry: DistanceRangeMap.RangeMapEntry<Speed>): Long {
+    return entry.value.millimetersPerSecond.toLong()
```

Comment:
isn't entry.value a long in java?

Reply:
In Java, if you have a `DistanceRangeMap<Speed>`, then the value of `DistanceRangeMap.RangeMapEntry<Speed> entry` is not converted to a long, it stays as Speed. Which is why it needs to be converted (and `toMetersPerSecond()` does not work, as Java expects a long as an argument, but would here get a Speed).